### PR TITLE
chore(go-module): migrate vendor to go module

### PR DESCRIPTION
**Pull request by vaniisgh**
wrt #1700

work in progress but would like some review :)
this my first PR as well here and my first time migrating code so just wanted to know if I am moving in the right direction.

using the command `GO111MODULE=on make all` package can be built successfully without being in the users $GOPATH

Signed-off-by: Vani Singh <vanisingh@live.co.uk>

also, I ran the tests on both my branches, some tests failed in both ... so I thought the migration was okay.
the tests that fail are :
- github.com/openebs/maya/cmd/maya-exporter/app/command
- github.com/openebs/maya/pkg/client/mapiserver

### CODE REVIEW

File: docs/developer.md [hunk 1]
```diff
@@ -131,9 +140,21 @@ $ cd $GOPATH/src/github.com/openebs/maya
...
 
+To build
+To build just standalone CLI for development:
+
  ```sh
- cd $GOPATH/src/github.com/openebs/maya
- make all
+
+ cd path/to/directory/openebs/maya
+ GO111MODULE=on make mayactl
```
Comment (mynktl) [inline docs/developer.md:149]: I think we can remove this. `go` will use go-module if go.mod file exists in root package.

Comment (vaniisgh) [reply docs/developer.md:149]: true, it does seem a little redundant :) will fix it

**Comment by mynktl** [comment]
Hi @vaniisgh 
Thanks for the PR!

There is one check for vendor dependency using `dep check` in `./buildscripts/travis-build.sh`. Need to remove that check.

**Comment by vaniisgh** [comment]
@mynktl There's this other little [issue](https://github.com/alecthomas/gometalinter/issues/562) with the make golint as well ... looking for a workaround. do you have any suggestions? :)

**Comment by vaniisgh** [comment]
I think we'll just need to find a different check for linting gometaliner doesn't have support for gomod. seemed best to remove it :)

**Comment by vaniisgh** [comment]
so, I ran the make kubegen, and it looks for the vendor/k8s.io/code-generator hence the revert.

**Comment by mynktl** [comment]
> so, I ran the `make kubegen`, and it looks for the vendor/k8s.io/code-generator hence the revert.

No need to add vendor code @vaniisgh . 

`GNUmakefile` and `buildscripts/code-gen.sh` uses code-generator from vendor. Since we are removing vendor code, you need to use different approach to install these binaries.

For buildscripts/code-gen.sh, you need to apply below patch....
```
diff --git a/buildscripts/code-gen.sh b/buildscripts/code-gen.sh
index 0cdb74eb0..7d76ef4ae 100755
--- a/buildscripts/code-gen.sh
+++ b/buildscripts/code-gen.sh
@@ -50,11 +50,14 @@ GROUPS_WITH_VERSIONS="$4"
 shift 4
 
 (
-  # To support running this script from anywhere, we have to first cd into this directory
-  # so we can install the tools.
-  #cd $(dirname "${0}")
-  cd vendor/k8s.io/code-generator/ 
-  go install ./cmd/{defaulter-gen,client-gen,lister-gen,informer-gen,deepcopy-gen}
+  # List of tools used for code generation
+  TOOL=(client-gen deepcopy-gen lister-gen informer-gen)
+  for i in ${TOOL[@]}; do
+    if [ ! -f "${GOPATH}/bin/$i" ]; then
+      echo "$i binary not found, installing at ${GOPATH}/bin"
+      go get k8s.io/code-generator/cmd/$i
+    fi
+  done
 )
 
 function codegen::join() { local IFS="$1"; shift; echo "$*"; }
```

For GNUmakefile you can replace 
`go install ./vendor/k8s.io/code-generator/cmd/<deepcopy-gen,client-gen,lister-gen,informer-gen>` 
with
`go get ./vendor/k8s.io/code-generator/cmd/<deepcopy-gen,client-gen,lister-gen,informer-gen>`.

Please let me know if you are having any issues.

**Comment by vaniisgh** [comment]
thanks for the help, I just panicked when the code broke tbh and though I'd just put them back in, tbh I don't think it solved much! will give this a go (and will be sure to bother you for future issues )

**Comment by vaniisgh** [comment]
closing this pull request because it was quite messy for me to revert the commits and sign the DCO, I hope thats ok :)
opened a fresh one with the same name. (#1702)

also, thank you for letting me make these changes even though you had it figured out it really helped me learn :) 

regards,
vani

**Pull request closed without merging**